### PR TITLE
Variables: Adds change tracking

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
First naive implementation of change tracking for variables. 

For more clarity we could create a separate reducer for the change tracking and add extraReducers for both transaction and sharedReducer. I choose to add isDirty to the transaction instead.

**Which issue(s) this PR fixes**:
Relates #39671

**Special notes for your reviewer**:

### CODE REVIEW

**Review by joshhunt — APPROVED** [review]
looks good :)

**Review by ashharrison90 — APPROVED** [review]
lgtm too 🙌

**Review by kaydelaney — APPROVED** [review]

File: public/app/features/variables/state/transactionReducer.ts [hunk 1]
```diff
@@ -32,10 +46,29 @@ const transactionSlice = createSlice({
...
 
+function actionAffectsDirtyState(action: AnyAction): boolean {
+  return [
+    removeVariable.type,
+    addVariable.type,
+    changeVariableProp.type,
+    changeVariableOrder.type,
+    duplicateVariable.type,
+    changeVariableType.type,
+    setCurrentVariableValue.type,
```
Comment (torkelo) [inline public/app/features/variables/state/transactionReducer.ts:68]: Not sure changing value should make it dirty, if I remember correctly the current diff change tracker ignores current value ?

**Review by torkelo — APPROVED** [review]
Looks good thanks for helping with this part for that change tracking Pr , got a bit stuck on this part

File: public/app/features/variables/state/transactionReducer.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/variables/state/transactionReducer.ts:68]: Good point I'll remove it, if the need arises it's easy to add back.

**Pull request merged**